src/System.Linq/tests/ConcatTests.cs

```diff
@@ -4,68 +4,48 @@
 
 using System;
 using System.Collections.Generic;
+using System.Diagnostics;
 using Xunit;
 
 namespace System.Linq.Tests
 {
     public class ConcatTests : EnumerableTests
     {
-        [Fact]
-        public void SameResultsRepeatCallsIntQuery()
-        {
-            var q1 = from x1 in new int?[] { 2, 3, null, 2, null, 4, 5 }
-                     select x1;
-            var q2 = from x2 in new int?[] { 1, 9, null, 4 }
-                     select x2;
-
-            Assert.Equal(q1.Concat(q2), q1.Concat(q2));
-        }
-
-        [Fact]
-        public void SameResultsRepeatCallsStringQuery()
-        {
-            var q1 = from x1 in new[] { "AAA", String.Empty, "q", "C", "#", "!@#$%^", "0987654321", "Calling Twice" }
-                     select x1;
-            var q2 = from x2 in new[] { "!@#$%^", "C", "AAA", "", "Calling Twice", "SoS" }
-                     select x2;
-
-            Assert.Equal(q1.Concat(q2), q1.Concat(q2));
-        }
+        private delegate IEnumerable<T> EnumerableTransform<T>(IEnumerable<T> input);
```

Comment:
I found this more readable-- having a `List<Func<IEnumerable<T>, IEnumerable<T>>>` seems quite verbose.